addons/mail/static/src/components/file_uploader/file_uploader.js

```diff
@@ -177,9 +177,10 @@ class FileUploader extends Component {
                         attachment.filename === filename &&
                         attachment.isTemporary
                     );
-                for (const attachment of relatedTemporaryAttachments) {
-                    attachment.delete();
-                }
+                if (relatedTemporaryAttachments) {
+                	for (const attachment of relatedTemporaryAttachments) {
+	                    attachment.delete();
+	                }
```

Comment:
Both indentation and scoping seem completely busted here, this change introduces tabs (not correct) and an outer `if () {` statement with no closing brace.

Also alternative less troublesome change: just use `relatedTemporaryAttachments || []`.

Also I can't help but find the use of `find` in the original code dodgy: according to the docstring in model.js it's supposed to return one record (or nothing), how does the iteration work?

Reply:
Indeed this is really strange, I'll have to check in more details before approving the PR.

Reply:
> So yeah the original code is just wrong, I guess the error was never handled properly before.
> 
> We could replace `find` by `all` to fix the loop issue, but even then the `=== filename` in the filter is wrong too as `filesData` just contains the error string in that case and not the actual file(name), so no attachment will ever be found (in other words that part of the code is useless). The whole deconstruct at line 169 is wrong for this case.
> 
> This code has been cleaned in master, but I think the easiest for now is just to delete it. Pending attachment won't be automatically deleted from the UI on error in this version, not a big deal.

good idea.